File: app/src/main/java/org/watsi/uhp/listeners/RemoveEncounterItemListener.java
package org.watsi.uhp.listeners;

import android.view.View;

import org.watsi.uhp.adapters.EncounterItemAdapter;
import org.watsi.uhp.models.Encounter;
import org.watsi.uhp.models.EncounterItem;

public class RemoveEncounterItemListener implements View.OnClickListener {

    private final Encounter mEncounter;
    private final EncounterItem mEncounterItem;
    private final EncounterItemAdapter mEncounterItemAdapter;

    public RemoveEncounterItemListener(
            Encounter encounter, EncounterItem encounterItem, EncounterItemAdapter adapter) {
        this.mEncounter = encounter;
        this.mEncounterItem = encounterItem;
        this.mEncounterItemAdapter = adapter;
    }

    @Override
    public void onClick(View v) {
        mEncounterItemAdapter.remove(mEncounterItem);
        mEncounter.removeEncounterItem(mEncounterItem);
    }
}

File: app/src/main/java/org/watsi/uhp/listeners/RemoveEncounterItemListener.kt
package org.watsi.uhp.listeners

import android.view.View
import org.watsi.domain.entities.Encounter
import org.watsi.domain.entities.EncounterItem

import org.watsi.uhp.adapters.EncounterItemAdapter

class RemoveEncounterItemListener(
        private val mEncounter: Encounter,
        private val mEncounterItem: EncounterItem,
        private val mEncounterItemAdapter: EncounterItemAdapter) : View.OnClickListener {

    override fun onClick(v: View) {
        mEncounterItemAdapter.remove(mEncounterItem)
        mEncounter.removeEncounterItem(mEncounterItem)
    }
}
